src/main/java/com/uber/jenkins/phabricator/coverage/CoberturaXMLParser.java

```diff
@@ -118,6 +118,21 @@ public InputSource resolveEntity(String publicId, String systemId) throws SAXExc
             NodeList classes = entry.getKey();
             List<String> sourceDirs = entry.getValue();
 
+            // Collect all filenames in coverage report
```

Comment:
We can actually make this lot more efficient

Phab already knows the files affected as part of the diff. In the same spirit as https://github.com/uber/phabricator-jenkins-plugin/pull/151

We can just collect file names from whats modified and just check state on those files and ignore the rest of the files in the coverage report. That will mitigate any large memory concerns as well

Reply:
@kageiit - if we check only files modified won't we miss files which were covered by new tests added into code base? Eg. if I simply add add new unit test which tests some real code but not changing that code itself - we will not even calculate line coverage?

Reply:
We dont need to calculate the coverage for the main file. We just need aggregares to report back which we can still do without looking if the file exists. Processing line coverage for every line will still be useless as we cannot show it in the ui